packages/web3-eth-accounts/src/index.js

```diff
@@ -129,7 +129,7 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
         error = false,
         transactionOptions = {},
         result,
-        hasTxSigningOptions = !!((tx.chain && tx.hardfork) || tx.common);
+        hasTxSigningOptions = !!(tx && ((tx.chain && tx.hardfork) || tx.common));
```

Comment:
Fixes small bug introduced in #3143. The test for whether or not tx is defined happens after this list of variables is set.